api/saml/configuration/model.py

```diff
@@ -64,6 +65,67 @@ class SAMLConfiguration(ConfigurationGrouping):
         format="narrow",
     )
 
+    patron_id_use_name_id = ConfigurationMetadata(
+        key="saml_patron_id_use_name_id",
+        label=_("Patron ID: Use SAML NameID"),
+        description=_(
+            "Boolean value indicating whether SAML NameID should be searched for a unique patron ID: "
+            "<br>"
+            "- <b>0</b> means that NameID won't be used, "
```

Comment:
@leonardr, @tdilauro, @jonathangreen, how do you think we need to represent boolean configuration settings in the admin UI and in the code?
- Is it enough to use an integer text input and treat them as binary zeros and ones? 
- Or maybe it would be better to represent them using a dropdown box containing "Yes", "No" choices?

P.S. Radio buttons would work nicely here, but, unfortunately, we don't have them in the admin UI yet.

Reply:
I like Yes/No, since they make more sense to humans. That would prevent nonsensical values and you could just set the current/default value, allowing us to omit text explaining the default.

Reply:
Take a look at core/config.py to see how we currently handle this.

```
        {
            "key": ALLOW_HOLDS,
            "label": _("Allow books to be put on hold"),
            "type": "select",
            "options": [
                { "key": "true", "label": _("Allow holds") },
                { "key": "false", "label": _("Disable holds") },
            ],
            "default": "true",
            "category": "Loans, Holds, & Fines",
            "level": SYS_ADMIN_ONLY
        },
```

The user interface looks good and it doesn't rely on people remembering constants. The dropdown format (rather than a checkbox) lets you explain each option individually. You can use `ConfigurationSetting.bool_value` to coerce the string value stored in the database (whatever it is) into a boolean.

Reply:
I updated the PR and turned `Patron ID: SAML NameID` setting into a boolean configuration setting, dropdown allowing admins to select one of the two values:
- `Use SAML NameID`
- `Do NOT use SAML NameID`